contracts/protocol/facets/DisputeResolverHandlerFacet.sol

```diff
@@ -188,40 +193,184 @@ contract DisputeResolverHandlerFacet is IBosonAccountEvents, ProtocolBase {
+
+        for (uint256 i = 0; i < _fieldsToUpdate.length; i++) {
+            DisputeResolverUpdateFields role = _fieldsToUpdate[i];
+
+            // Approve admin update
+            if (role == DisputeResolverUpdateFields.Admin && disputeResolverPendingUpdate.admin != address(0)) {
+                require(disputeResolverPendingUpdate.admin == sender, UNAUTHORIZED_CALLER_UPDATE);
+
+                preUpdateDisputeResolverCheck(_disputeResolverId, sender, lookups);
+
+                // Delete old disputeResolver id by admin mapping
+                delete lookups.disputeResolverIdByAdmin[disputeResolver.admin];
+
+                // Update admin
+                disputeResolver.admin = sender;
+
+                // Store new disputeResolver id by admin mapping
+                lookups.disputeResolverIdByAdmin[sender] = _disputeResolverId;
+
+                // Delete pending update admin
+                delete disputeResolverPendingUpdate.admin;
+            }
+
+            // Approve operator update
+            if (role == DisputeResolverUpdateFields.Operator && disputeResolverPendingUpdate.operator != address(0)) {
+                require(disputeResolverPendingUpdate.operator == sender, UNAUTHORIZED_CALLER_UPDATE);
+
+                preUpdateDisputeResolverCheck(_disputeResolverId, sender, lookups);
+
+                // Delete old disputeResolver id by operator mapping
+                delete lookups.disputeResolverIdByOperator[disputeResolver.operator];
+
+                // Update operator
+                disputeResolver.operator = sender;
+
+                // Store new disputeResolver id by operator mapping
+                lookups.disputeResolverIdByOperator[sender] = _disputeResolverId;
+
+                // Delete pending update operator
+                delete disputeResolverPendingUpdate.operator;
+            }
+
+            // Aprove clerk update
+            if (role == DisputeResolverUpdateFields.Clerk && disputeResolverPendingUpdate.clerk != address(0)) {
+                require(disputeResolverPendingUpdate.clerk == sender, UNAUTHORIZED_CALLER_UPDATE);
+
+                preUpdateDisputeResolverCheck(_disputeResolverId, sender, lookups);
+
+                // Delete old disputeResolver id by clerk mapping
+                delete lookups.disputeResolverIdByClerk[disputeResolver.clerk];
+
+                // Update clerk
+                disputeResolver.clerk = sender;
+
+                // Store new disputeResolver id by clerk mapping
+                lookups.disputeResolverIdByClerk[sender] = _disputeResolverId;
+
+                // Delete pending update clerk
+                delete disputeResolverPendingUpdate.clerk;
+            }
+        }
 
         // Notify watchers of state change
-        emit DisputeResolverUpdated(_disputeResolver.id, _disputeResolver, sender);
+        emit DisputeResolverUpdateApplied(_disputeResolverId, disputeResolver, disputeResolverPendingUpdate, sender);
```

Comment:
One observation - it can happend that actually nothing is updated, but we still emit this event.
Yes, we do check in line 306 that some pending update exists. But that is true is some some fields is not zero.

Still it can happen that let's say `sellerPendingUpdate.admin` is not zero but operator and treasury are. So check in 306 will pass. However, I might put in ` _fieldsToUpdate` only `SellerUpdateFields.Operator` which is not supposed to be updated. But tx will be confirmed normally, so whoever called it they might think that operator update was succeful (if they don't verify the emitted log).

I think we should track that some update really took place and emit the even only if it did and return otherwise.

And I think we have the same situation in `optInToSellerUpdate`

Reply:
done!